src/Compilers/CSharp/Test/Syntax/Parsing/WithElementParsingTests.cs

```diff
@@ -0,0 +1,2336 @@
+                N(SyntaxKind.CommaToken);
+                N(SyntaxKind.Argument);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "b");
+                    }
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void WithElement10(LanguageVersion languageVersion)
+    {
+        UsingExpression("[with(), with()]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            CollectionArgumentsOrInvocation(languageVersion);
+            N(SyntaxKind.ArgumentList);
+            {
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CommaToken);
+            CollectionArgumentsOrInvocation(languageVersion);
+            N(SyntaxKind.ArgumentList);
+            {
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void WithElement11(LanguageVersion languageVersion)
+    {
+        UsingExpression("[a, with()]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.ExpressionElement);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "a");
+                }
+            }
+            N(SyntaxKind.CommaToken);
+            CollectionArgumentsOrInvocation(languageVersion);
+            N(SyntaxKind.ArgumentList);
+            {
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Theory, MemberData(nameof(CollectionArgumentsLanguageVersions))]
+    public void WithElement12(LanguageVersion languageVersion)
+    {
+        UsingExpression("[a:b, with()]",
+            TestOptions.Regular.WithLanguageVersion(languageVersion),
+            // (1,3): error CS1003: Syntax error, ',' expected
+            // [a:b, with()]
+            Diagnostic(ErrorCode.ERR_SyntaxError, ":").WithArguments(",").WithLocation(1, 3),
+            // (1,4): error CS1003: Syntax error, ',' expected
+            // [a:b, with()]
+            Diagnostic(ErrorCode.ERR_SyntaxError, "b").WithArguments(",").WithLocation(1, 4));
+
+        if (languageVersion == LanguageVersion.CSharp14)
```

Comment:
nit: it looks like this test could be condensed to use 'CollectionARgumentsOrInvocation' helper.

Reply:
f2c6c4dee0a